### PR TITLE
Feat/eslint

### LINKED ISSUE #378: Add eslint to improve code consistency
We should add eslint to allow much easier debugging in editors as well as consistent formatting and code style throughout the codebase.

The Airbnb eslint config is a popular and good option for react, but is notably strict and may introduce some friction for newcomers.

The Standard + Prettier configs are also pretty good, but aren't as strict, so they might be more beginner friendly.

### PR DESCRIPTION
**Pull request by johnpyp**
## Description

Adds eslint to consistently format codebase and enforce code quality in various areas.

Fixes #378 

## Motivation and Context

There was inconsistent code style and some bad practices throughout the codebase. This change attempts to solve that with stricter linting and formatting requirements.

## Type of change

- [ x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by johnpyp** [comment]
Up to date with the current development branch.

Includes:
- Airbnb eslint config
- Prettier with eslint integration (via prettier eslint config)
- All components now have type validation (enforced by eslint)
- Various package upgrades, such as react-scripts, react and redux
... and other more minor changes

Other changes that could either go into this PR or a new one:
- Cleaning up naming, such as the issue right now where `resource` describes many different things, including the plural (array) and singular version of it.
- Upgrading the rest of the packages to, ideally, the latest version
- Cleaning up messy or bad code patterns
- Using functional components when possible

File: src/components/NotFoundPage/NotFoundPageLayout.js [hunk 1]
```diff
@@ -12,3 +12,5 @@ export const NotFoundPageLayout = styled('div')`
   right: 0;
   margin: auto;
 `;
+
+export default NotFoundPageLayout;
```
Comment (davidholyko) [inline src/components/NotFoundPage/NotFoundPageLayout.js]: looks like i forgot to delete this file. you can go ahead and remove it and remove the dependencies as well

File: src/components/SavedResources/SavedResourcesContainer.js [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
+    removeItem: null
+  };
+
   constructor(props) {
     super(props);
     this.state = {
-      data: Object.assign([], this.props.data),
+      data: Object.assign([], this.props.data)
     };
     this.onDragEnd = this.onDragEnd.bind(this);
```
Comment (davidholyko) [inline src/components/SavedResources/SavedResourcesContainer.js]: we can turn this into an arrow function and removing the binding

**Comment by davidholyko** [comment]
i think its good to go. there are some small things here and there that we can break into smaller PRs for the future but this is a big step in the right direction

**Comment by johnpyp** [comment]
Awesome. It would be cool if we could get this merged into development before the next meeting, because other changes or PRs made before then will have a lot of merge conflicts to deal with.

**Comment by davidholyko** [comment]
i would ping raj and have him approve/merge it in

**Comment by johnpyp** [comment]
Sounds good. @RajChaudhry Do you think this is good to merge?

**Comment by johnpyp** [comment]
Added lint-staged and husky to fix #388

**Comment by johnpyp** [comment]
Added trailing commas to prettier config (already present in airbnb config, but was overridden by the prettier eslint plugin).

**Comment by RajChaudhry** [comment]
Shit forgot to write comments before merging, but the linting looks great! Thanks for setting it up JP!

**Pull request merged**